### PR TITLE
[#3732] Update Python client to align with the Java client API

### PR DESCRIPTION
**Pull request by LanceHsun**
### What changes were proposed in this pull request?

- Refactored the `SupportsSchemas.list_schemas` method to return a list of strings (`List[str]`) instead of `NameIdentifier` objects.
- Updated the `BaseSchemaCatalog` class and related implementations to reflect this change.
- Modified relevant test cases to verify the new return type and ensure functionality.

### Why are the changes needed?

- The refactoring aligns the `list_schemas` method with updated API specifications, making it more user-friendly by returning schema names as strings.
- 
Fix: #3701
### Does this PR introduce _any_ user-facing change?

Yes, this PR introduces the following user-facing changes:
- The `list_schemas` method in the `SupportsSchemas` interface now returns a list of schema names as strings (`List[str]`) instead of a list of `NameIdentifier` objects.

### How was this patch tested?

- Modified existing unit tests to cover the changes in the `list_schemas` method.

### CODE REVIEW

File: clients/client-python/gravitino/api/supports_schemas.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ class SupportsSchemas(ABC):
     """
 
     @abstractmethod
-    def list_schemas(self, namespace: Namespace) -> List[NameIdentifier]:
```
Comment (shaofengshi) [inline clients/client-python/gravitino/api/supports_schemas.py:28]: In the new API, this method doesn't have any input parameter (like the namespace). So the namespace information should come from the BaseSchemaCatalog itself. But now the BaseSchemaCatalog object doesn't hold the namespace information, so you need to pass-in that information in the init method. You can refer to the Java change at commit 2687f1aa1819fe301151dded82563911875ddf62

File: clients/client-python/gravitino/catalog/base_schema_catalog.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def list_schemas(self, namespace: Namespace) -> [NameIdentifier]:
             resp.body, infer_missing=True
         )
         entity_list_response.validate()
-        return entity_list_response.identifiers()
+        return [str(identifier) for identifier in entity_list_response.identifiers()]
 
     def create_schema(
```
Comment (shaofengshi) [inline clients/client-python/gravitino/catalog/base_schema_catalog.py:83]: The "createSchema", "loadSchema", "alterSchema" and "dropSchema" all need be changed to use a string parameter to replace the NameIdentifier parameter.

**Comment by jerryshao** [comment]
@noidname01 would you please help to review this?

**Comment by noidname01** [comment]
@jerryshao Sure

File: clients/client-python/gravitino/catalog/base_schema_catalog.py (hunk 2 above)
Comment (LanceHsun) [reply clients/client-python/gravitino/catalog/base_schema_catalog.py:83]: done!

File: clients/client-python/gravitino/api/supports_schemas.py (hunk 1 above)
Comment (shaofengshi) [reply clients/client-python/gravitino/api/supports_schemas.py:28]: the interface not changed?

File: clients/client-python/gravitino/catalog/base_schema_catalog.py [hunk 3]
```diff
@@ -53,43 +54,42 @@ def __init__(
...
         resp = self.rest_client.get(
-            BaseSchemaCatalog.format_schema_request_path(namespace)
+            BaseSchemaCatalog.format_schema_request_path(self.schema_namespace)
         )
         entity_list_response = EntityListResponse.from_json(
             resp.body, infer_missing=True
         )
         entity_list_response.validate()
-        return entity_list_response.identifiers()
+        return [str(identifier) for identifier in entity_list_response.identifiers()]
```
Comment (shaofengshi) [inline clients/client-python/gravitino/catalog/base_schema_catalog.py:81]: I'm not sure whether the return value is an array of schema name, or an array of identifier's toString() value; Please double check

**Comment by shaofengshi** [comment]
Please check the python integration errors below, you need to fix them @LanceHsun :

************* Module gravitino.dto.dto_converters
gravitino/dto/dto_converters.py:42:19: E1120: No value for argument 'namespace' in constructor call (no-value-for-parameter)
************* Module tests.integration.test_fileset_catalog
tests/integration/test_fileset_catalog.py:90:16: E1123: Unexpected keyword argument 'ident' in method call (unexpected-keyword-arg)
tests/integration/test_fileset_catalog.py:90:16: E1120: No value for argument 'schema_name' in method call (no-value-for-parameter)
tests/integration/test_fileset_catalog.py:119:8: E1123: Unexpected keyword argument 'ident' in method call (unexpected-keyword-arg)
tests/integration/test_fileset_catalog.py:119:8: E1120: No value for argument 'schema_name' in method call (no-value-for-parameter)
tests/integration/test_fileset_catalog.py:150:45: E1123: Unexpected keyword argument 'namespace' in method call (unexpected-keyword-arg)
************* Module tests.integration.test_schema
tests/integration/test_schema.py:112:15: E1123: Unexpected keyword argument 'ident' in method call (unexpected-keyword-arg)
tests/integration/test_schema.py:112:15: E1120: No value for argument 'schema_name' in method call (no-value-for-parameter)
tests/integration/test_schema.py:129:12: E1123: Unexpected keyword argument 'ident' in method call (unexpected-keyword-arg)
tests/integration/test_schema.py:129:12: E1120: No value for argument 'schema_name' in method call (no-value-for-parameter)
tests/integration/test_schema.py:135:33: E1123: Unexpected keyword argument 'namespace' in method call (unexpected-keyword-arg)
tests/integration/test_schema.py:143:17: E1123: Unexpected keyword argument 'ident' in method call (unexpected-keyword-arg)
tests/integration/test_schema.py:143:17: E1120: No value for argument 'schema_name' in method call (no-value-for-parameter)

File: clients/client-python/gravitino/catalog/base_schema_catalog.py (hunk 3 above)
Comment (LanceHsun) [reply clients/client-python/gravitino/catalog/base_schema_catalog.py:81]: The Python method list_schemas now returns an array of schema names. The change [str(identifier) for identifier in entity_list_response.identifiers()] ensures that it matches the Java method listSchemas, which also returns an array of schema names by invoking NameIdentifier::name.

**Comment by jerryshao** [comment]
@noidname01 would you please help to work on this? Thanks a lot. I think @LanceHsun may not have the time to continue the work.

**Comment by noidname01** [comment]
> @noidname01 would you please help to work on this? Thanks a lot. I think @LanceHsun may not have the time to continue the work.

Sure, let me take a look.

**Comment by shaofengshi** [comment]
These changes have been merged by Tim in another PR, so close this one.

**Pull request closed without merging**